### PR TITLE
git lfs status --json only includes lfs files

### PR DESCRIPTION
**Pull request by asottile**
More from #2289

### CODE REVIEW

File: commands/command_status.go [hunk 1]
```diff
@@ -244,9 +244,25 @@ func jsonStagedPointers(ref string) {
 		ExitWithError(err)
 	}
 
+	scanner, err := lfs.NewPointerScanner()
```
Comment (ttaylorr) [inline commands/command_status.go]: It looks like this is being created in order to pass it to the `blobInfoFrom` call on L257 below. This is fine, though I'd rather pass the `*lfs.PointerScanner` we have in the `statusCommand` top-level function, like this:

```diff
diff --git a/commands/command_status.go b/commands/command_status.go
index e74e26b5..bfda7f8e 100644
--- a/commands/command_status.go
+++ b/commands/command_status.go
@@ -30,11 +30,18 @@ func statusCommand(cmd *cobra.Command, args []string) {
 		scanIndexAt = git.RefBeforeFirstCommit
 	}
 
+	scanner, err := lfs.NewPointerScanner()
+	if err != nil {
+		scanner.Close()
+
+		ExitWithError(err)
+	}
+
 	if porcelain {
 		porcelainStagedPointers(scanIndexAt)
 		return
 	} else if statusJson {
-		jsonStagedPointers(scanIndexAt)
+		jsonStagedPointers(scanner, scanIndexAt)
 		return
 	}
 
@@ -44,14 +51,6 @@ func statusCommand(cmd *cobra.Command, args []string) {
 	if err != nil {
 		ExitWithError(err)
 	}
-
-	scanner, err := lfs.NewPointerScanner()
-	if err != nil {
-		scanner.Close()
-
-		ExitWithError(err)
-	}
-
 	Print("\nGit LFS objects to be committed:\n")
 	for _, entry := range staged {
 		switch entry.Status {
@@ -238,7 +237,7 @@ type JSONStatus struct {
 	Files map[string]JSONStatusEntry `json:"files"`
 }
 
-func jsonStagedPointers(ref string) {
+func jsonStagedPointers(s *lfs.PointerScanner, ref string) {
 	staged, unstaged, err := scanIndex(ref)
 	if err != nil {
 		ExitWithError(err)
@@ -247,6 +246,9 @@ func jsonStagedPointers(ref string) {
 	status := JSONStatus{Files: make(map[string]JSONStatusEntry)}
 
 	for _, entry := range append(unstaged, staged...) {
+		_, from, err := blobInfoFrom(s, entry)
+		// ...
+
 		switch entry.Status {
 		case lfs.StatusRename, lfs.StatusCopy:
 			status.Files[entry.DstName] = JSONStatusEntry{
```

File: test/test-status.sh [hunk 2]
```diff
@@ -108,6 +108,13 @@ begin_test "status --json"
 
   expected='{"files":{"file2.dat":{"status":"R","from":"file1.dat"}}}'
   [ "$expected" = "$(git lfs status --json)" ]
+
+  git commit -m "file1.dat -> file2.dat"
+
+  echo hi > test1.txt
```
Comment (ttaylorr) [inline test/test-status.sh]: Thank you for adding this test case! Would you mind either:

- Adding a `#` comment above saying what this is testing, or...
- Extracting this out to a separate test?

**Comment by asottile** [comment]
Thanks!  Updated :D

**Comment by ttaylorr** [comment]
Thanks again for your contributions to LFS! I'll schedule this for v2.2.1.

**Comment by asottile** [comment]
Awesome!  Looking forward to 2.2.1 :D

**Pull request merged**